llvm/test/CodeGen/AArch64/qmovn.ll

```diff
@@ -292,15 +292,15 @@ entry:
 
 ; Test the (concat_vectors (X), (trunc(umin(smax(Y, 0), 2^n))))) pattern.
 
+; TODO: %min is a value between 0 and 255 and is within the unsigned range of i8.
+; So it is saturated truncate. we have an optimization opportunity.
```

Comment:
I'm pretty sure this should be matching a sqxtun2, that was the intent. I believe because the lower limit is already clamped, that the upper smin is equivalent to umin.
https://godbolt.org/z/e7ne31TYb
You can see in that example that the midend turned smin into umin.

Reply:
(right place :))
agree. I also added a comment with that intention and I'm thinking maybe adding code to DAGCombiner can solve it.
do you think I should include this in this patch as well?